modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/authenticator/LDAPAuth.java

```diff
@@ -375,48 +386,42 @@ private int _authenticate(
 			LDAPAuthResult ldapAuthResult = _authenticate(
 				safeLdapContext, companyId, attributes, fullUserDN, password);
 
-			// Get user or create fromUnsafe LDAP
-
 			if (!ldapAuthResult.isAuthenticated()) {
 				password = null;
-			}
 
-			User user = _ldapUserImporter.importUser(
-				ldapServerId, companyId, safeLdapContext, attributes, password);
-
-			if (!ldapAuthResult.isAuthenticated()) {
 				PasswordModificationThreadLocal.setPasswordModified(false);
-			}
 
-			// Process LDAP failure codes
+				// Process LDAP failure codes
 
-			String errorMessage = ldapAuthResult.getErrorMessage();
+				String errorMessage = ldapAuthResult.getErrorMessage();
 
-			if (errorMessage != null) {
-				SystemLDAPConfiguration systemLDAPConfiguration =
-					_systemLDAPConfigurationProvider.getConfiguration(
-						companyId);
+				if (errorMessage != null) {
+					SystemLDAPConfiguration systemLDAPConfiguration =
+						_systemLDAPConfigurationProvider.getConfiguration(
+							companyId);
 
-				for (String errorUserLockoutKeyword :
-						systemLDAPConfiguration.errorUserLockoutKeywords()) {
+					for (String errorUserLockoutKeyword :
+							systemLDAPConfiguration.
+								errorUserLockoutKeywords()) {
 
-					if (errorMessage.contains(errorUserLockoutKeyword)) {
-						throw new UserLockoutException.LDAPLockout(
-							fullUserDN, errorMessage);
+						if (errorMessage.contains(errorUserLockoutKeyword)) {
+							throw new UserLockoutException.LDAPLockout(
+								fullUserDN, errorMessage);
+						}
 					}
-				}
 
-				for (String errorPasswordExpiredKeyword :
-						systemLDAPConfiguration.
-							errorPasswordExpiredKeywords()) {
+					for (String errorPasswordExpiredKeyword :
+							systemLDAPConfiguration.
+								errorPasswordExpiredKeywords()) {
+
+						if (errorMessage.contains(
+								errorPasswordExpiredKeyword)) {
 
-					if (errorMessage.contains(errorPasswordExpiredKeyword)) {
-						throw new PasswordExpiredException();
+							throw new PasswordExpiredException();
+						}
 					}
 				}
```

Comment:
This part was previously not inside the `if (!ldapAuthResult.isAuthenticated())` condition. As I see the `errorMessage` is only getting set after `isAuthenticated` is set to false but have you validated that it is true and no other scenario can happen?

Reply:
Yes, I just reduced the amount of verifications as I noticed that this `!ldapAuthResult.isAuthenticated()` condition was being called several times sequentially in a small time interval.